src/pages/iou/request/step/withWritableReportOrNotFound.tsx

```diff
@@ -54,13 +51,18 @@ export default function <TProps extends WithWritableReportOrNotFoundProps<MoneyR
     shouldIncludeDeprecatedIOUType = false,
 ): React.ComponentType<Omit<TProps & RefAttributes<TRef>, keyof WithWritableReportOrNotFoundOnyxProps>> {
     // eslint-disable-next-line rulesdir/no-negated-variables
-    function WithWritableReportOrNotFound(props: TProps, ref: ForwardedRef<TRef>) {
-        const {report = {reportID: ''}, route, isLoadingApp = true, reportDraft} = props;
+    function WithWritableReportOrNotFound(props: Omit<TProps, keyof WithWritableReportOrNotFoundOnyxProps>, ref: ForwardedRef<TRef>) {
+        const {route} = props;
+        const [report] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${route.params.reportID ?? '-1'}`);
+        const [isLoadingApp = true] = useOnyx(ONYXKEYS.IS_LOADING_APP);
+        const [reportDraft] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_DRAFT}${route.params.reportID ?? '-1'}`);
```

Comment:
```suggestion
        const [report] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${props.route.params.reportID ?? '-1'}`);
        const [isLoadingApp = true] = useOnyx(ONYXKEYS.IS_LOADING_APP);
        const [reportDraft] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_DRAFT}${props.route.params.reportID ?? '-1'}`);
```

Reply:
@blazejkustra the route prop is being used in other places of the same component

Reply:
@brunovjk resolving this as we need to destructure route prop anyway for other instances